lib/glific/contacts/import.ex

```diff
@@ -115,6 +134,32 @@ defmodule Glific.Contacts.Import do
     end
   end
 
+  ## create contact fields from sheet header and the first contact
+  @spec create_contact_fields(list) :: list
+  defp create_contact_fields(contacts) do
+    first_contact = Enum.at(contacts, 0)
+
+    Enum.each(first_contact.contact_fields, fn {field, _value} ->
+      field = Glific.string_snake_case(field)
+
+      case Repo.get_by(Glific.Contacts.ContactsField, %{shortcode: field},
+             organization_id: first_contact.organization_id
+           ) do
+        nil ->
+          ContactField.create_contact_field(%{
+            name: field,
+            shortcode: field,
+            organization_id: first_contact.organization_id
+          })
+
+        contact_field ->
+          {:ok, contact_field}
+      end
+    end)
+
```

Comment:
@mdshamoon let's create a function in contacts table with name update_contact_fileds(contact, fields ) 

There are some inconsistencies in the flows also for creating contact fields we will migrate that to this centre function slowly.

Reply:
Didn't understand the contacts table?  

Do you mean I need to replace the `do_add_contact_fields` function with `update_contact_fields` function which should be created in contacts.ex file?